### ISSUE TITLE
Allow TypeAlias of Annotated

### ISSUE THREAD

**Issue by alanhdu** [python/mypy#16162]
**Feature**

This is a really small feature, but we would like to be able to use a TypeAlias for PEP 593 Annotated like:
```
NewName: TypeAlias = Annotated
```

**Pitch**

We have a class called `Annotations` to represent a piece of a data, and would like to attach schema information to them. We've had a couple questions about `Annotated[Annotations, Schema]`, and so would like to provide an alternative name that could be easier to read (potentially `WithSchema[Annotations, Schema]`.

In mypy, however, this raises an error:
```python
from typing_extensions import Annotated, TypeAlias

WithSchema: TypeAlias = Annotated

def f(x: WithSchema[int, 5]):
    pass
```
 
gives
 
 ```
main.py:3: error: Annotated[...] must have exactly one type argument and at least one annotation  [valid-type]
main.py:5: error: Bad number of arguments for type alias, expected: 0, given: 2  [type-arg]
main.py:5: error: Invalid type: try using Literal[5] instead?  [valid-type]
Found 3 errors in 1 file (checked 1 source file)
 ```
 
 This is a minor readability point, but how easy/hard would it be to enable this type of usage?

**Comment by AlexWaygood** [comment]
FWIW, if you use an aliased import, mypy will be cool with what you're trying to do:

```python
from typing_extensions import Annotated as WithSchema

def f(x: WithSchema[int, 5]):
    pass
```

I think we're probably unlikely to allow using TypeAlias like this, since the object on the r.h.s. _should_ always be valid as a type. `Annotated` by itself isn't valid as a type — it's a special form for constructing custom types (sometimes called a "typeform").
